app/javascript/controllers/registration_tabs_controller.js

```diff
@@ -2,8 +2,19 @@ import { Controller } from '@hotwired/stimulus'
 
 export default class extends Controller {
   static targets = ['requiredFormField', 'requiredCsvField']
+  static values = {
+    csv: Boolean
+  }
+
+  connect () {
+    // This check is used to automatically toggle the CSV-oriented registration
+    // form fields only at connect() time when the `csv` value has been set in
+    // the HTML, which only happens after the CSV form has been submitted. It is
+    // primarily used in situations where there are CSV upload errors.
```

Comment:
rewording suggestion. (I found it hard to grok what you wrote -- too many clauses?):

used only at connect() time when the `csv` value has been set in the HTML (which only occurs after the CSV form has been submitted).  Useful for CSV upload errors.

Reply:
Done